airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/up_to_date/pipeline.py

```diff
@@ -0,0 +1,322 @@
+                    step=self,
+                    status=StepStatus.SKIPPED,
+                    stderr=f"CDK is already up to date. Current version: {current_cdk_version}, requested version: {self.cdk_version}",
+                )
+
+            with_new_cdk = await connector_container.with_exec(["poetry", "add", f"airbyte-cdk=={self.cdk_version}"])
+            await with_new_cdk.file("pyproject.toml").export(f"{self.context.connector.code_directory}/pyproject.toml")
+            self.logger.info(f"Generated pyproject.toml for {self.context.connector.technical_name}")
+            await with_new_cdk.file("poetry.lock").export(f"{self.context.connector.code_directory}/poetry.lock")
+            self.logger.info(f"Generated poetry.lock for {self.context.connector.technical_name}")
+        except dagger.ExecError as e:
+            return StepResult(step=self, status=StepStatus.FAILURE, stderr=str(e))
+
+        return StepResult(step=self, status=StepStatus.SUCCESS)
+
+
+class RestoreOriginalState(Step):
+    context: ConnectorContext
+
+    title = "Restore original state"
+
+    def __init__(self, context: ConnectorContext) -> None:
+        super().__init__(context)
+        self.pyproject_path = context.connector.code_directory / "pyproject.toml"
+        if self.pyproject_path.exists():
+            self.original_pyproject = self.pyproject_path.read_text()
+        self.poetry_lock_path = context.connector.code_directory / "poetry.lock"
+        if self.poetry_lock_path.exists():
+            self.original_poetry_lock = self.poetry_lock_path.read_text()
+
+    async def _run(self) -> StepResult:
+        if self.original_pyproject:
+            self.pyproject_path.write_text(self.original_pyproject)
+            self.logger.info(f"Restored pyproject.toml for {self.context.connector.technical_name}")
+        if self.original_poetry_lock:
+            self.poetry_lock_path.write_text(self.original_poetry_lock)
+            self.logger.info(f"Restored poetry.lock for {self.context.connector.technical_name}")
+
+        return StepResult(
+            step=self,
+            status=StepStatus.SUCCESS,
+        )
+
+
+class RegressionTest(Step):
+    """Run the regression test for the connector.
+    We test that:
+    - The original dependencies are installed in the new connector image.
+    - The dev dependencies are not installed in the new connector image.
+    - The connector spec command successfully.
+    """
+
+    context: ConnectorContext
+
+    title = "Run regression test"
+
+    async def _run(self, new_connector_container: dagger.Container) -> StepResult:
+        try:
+            await new_connector_container.with_exec(["spec"])
+            await new_connector_container.with_mounted_file(
+                "pyproject.toml", (await self.context.get_connector_dir(include=["pyproject.toml"])).file("pyproject.toml")
+            ).with_exec(["poetry", "run", self.context.connector.technical_name, "spec"], skip_entrypoint=True)
```

Comment:
Are we mounting these files because they are not kept after build on the built connector?

Reply:
I don't know anything about mounting. I was just copying what you did in the migrate_to_poetry action. Is this what it means: You should mount things that might have changed since the container was made ?